templates/deployment.yaml

```diff
@@ -81,22 +88,61 @@ spec:
 {{ toYaml .Values.resources | indent 12 }}
         volumeMounts:
         - name: xwiki-data
+          mountPath: /usr/local/xwiki
+        - name: xwiki-solr-data
           mountPath: /opt/solr/server/solr
+        - name: xwiki-tomcat-conf
+          mountPath: /usr/local/tomcat/conf/web.xml
+          subPath: web.xml
       volumes:
+      - name: xwiki-solr-data
+        persistentVolumeClaim:
+          claimName: xwiki-solr
       - name: xwiki-data
         persistentVolumeClaim:
           claimName: xwiki
+      - name: xwiki-tomcat-conf
+        configMap:
+          name: xwiki-tomcat-conf
+          items:
+          - key: web.xml
+            path: web.xml
+
 ---
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   name: xwiki
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
 spec:
   accessModes:
-    - ReadWriteMany
+    - {{.Values.storageAccessMode}}
   resources:
     requests:
       storage: {{ .Values.storage }}
     {{- if .Values.storageClass }}
     storageClassName: {{ .Values.storageClass | quote }}
     {{- end }}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: xwiki-solr
```

Comment:
Hardcoded name, makes it impossible to install two versions on the same namespace.

Reply:
Other places in the original chart hardcode the resource names. For now, I'll just keep it as it is

Reply:
Are you not going to change this? Usually something like `name: {{ include "my-chart.fullName" . }}-xwiki-solr` will be better. See other charts as an example. [This is one file for MariaDB](https://github.com/bitnami/charts/blob/master/bitnami/mariadb/templates/primary/configmap.yaml), all of them are similar.